### ISSUE TITLE
Gym gets stuck signing (First time submissions only) (reproducible)

### ISSUE THREAD

**Issue by EricNetsch** [fastlane/fastlane#10108]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I have set up a relatively stable fastlane process on Bitrise which has the ability to submit multiple  apps across a range of Apple Developer accounts. 

Every-time I  deploy an app for the first time on Apple Developer and  ITC gym will get stuck signing my Notification.apex extension. When I rerun the exact same build a second time, gym works without any issues. This is reproducible where the first build of an app always gets stuck and times-out.

I believe this has something to do with Match creating the provisioning profiles and installing them for the first time vs. when the already exist in my Match repo the second time around. 

Some notes on the project: 
-Dynamic Apple Developer accounts and ITC (personal and organization)
-Multiple Targets (1 for the app and 1 for notification extension)
-Creating custom keychain and destroying it after each build before match and after final deliver.
- passing custom xcarg to properly codesign the app and notification extension target properly.

Please let me know what I can do test/troubleshoot to resolve this issue.

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

```
+------------------------------------------------------------------------------+

| (4) fastlane                                                                 |
+------------------------------------------------------------------------------+
| id: fastlane                                                                 |
| version: 2.3.9                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2017-08-22T11:23:22-07:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |
INFO[11:23:22]  * [OK] Step dependency (go) installed, available. 
Configs:
- WorkDir: /Users/vagrant/git
- Lane: create
- UpdateFastlane: true
Expand WorkDir
Expanded WorkDir: /Users/vagrant/git
Determine desired Fastlane version
Checking Gemfile.lock (/Users/vagrant/git/Gemfile.lock) for fastlane gem
Gemfile.lock does not exist
Update system installed Fastlane
$ gem "install" "fastlane" "--no-document"
Successfully installed commander-fastlane-4.4.5
Successfully installed declarative-0.0.9
Successfully installed declarative-option-0.1.0
Successfully installed representable-3.0.4
Successfully installed mime-types-data-3.2016.0521
Successfully installed mime-types-3.1
Successfully installed google-api-client-0.12.0
Successfully installed fastlane-2.53.1
8 gems installed
Fastlane version:
$ fastlane "--version"
fastlane installation at path:
/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/bin/fastlane
-----------------------------
fastlane 2.53.1
Run Fastlane
$ fastlane "create"
[11:23:34]: For more information, check out https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[11:23:37]: -------------------------------------------------
[11:23:37]: --- Step: Verifying required fastlane version ---
[11:23:37]: -------------------------------------------------
[11:23:37]: Your fastlane version 2.53.1 matches the minimum requirement of 1.109.0  ✅
[11:23:37]: ------------------------------
[11:23:37]: --- Step: default_platform ---
[11:23:37]: ------------------------------
[11:23:37]: ---------------------
[11:23:37]: --- Step: team_id ---
[11:23:37]: ---------------------
[11:23:37]: Setting Team ID to 'XXXXXXXX' for all build steps
[11:23:37]: Driving the lane 'create' 🚀
[11:23:37]: ----------------------------------
[11:23:37]: --- Step: set_info_plist_value ---
[11:23:37]: ----------------------------------
[11:23:37]: ----------------------------------
[11:23:37]: --- Step: set_info_plist_value ---
[11:23:37]: ----------------------------------
[11:23:37]: ----------------------------------
[11:23:37]: --- Step: set_info_plist_value ---
[11:23:37]: ----------------------------------
[11:23:37]: ----------------------------------
[11:23:37]: --- Step: set_info_plist_value ---
[11:23:37]: ----------------------------------
[11:23:37]: ----------------------------------
[11:23:37]: --- Step: set_info_plist_value ---
[11:23:37]: ----------------------------------
[11:23:37]: ----------------------------------
[11:23:37]: --- Step: set_info_plist_value ---
[11:23:37]: ----------------------------------
[11:23:37]: ----------------------------------
[11:23:37]: --- Step: set_info_plist_value ---
[11:23:37]: ----------------------------------
[11:23:37]: ---------------------
[11:23:37]: --- Step: produce ---
[11:23:37]: ---------------------
+------------------------------------+----------------------------------------------------+
|                               Summary for produce 2.53.1                                |
+------------------------------------+----------------------------------------------------+
| username                           | xxx@tapcart.co                                    |
| app_identifier                     | com.tapcart.XXXXXXXX                             |
| app_name                           | Timeless Skin Care - Buy Products Through Our App! |
| language                           | English                                            |
| app_version                        | 1.0                                                |
| sku                                | XXXXXXXX                                         |
| itc_team_name                      | Company, LLC                                       |
| company_name                       | Company, LLC                                       |
| enable_services.apple_pay          | on                                                 |
| enable_services.associated_domains | on                                                 |
| enable_services.push_notification  | on                                                 |
| platform                           | ios                                                |
| skip_itc                           | false                                              |
| skip_devcenter                     | false                                              |
| team_id                            | XXXXXXXX                                         |
+------------------------------------+----------------------------------------------------+
[11:23:39]: Creating new app 'Timeless Skin Care - Buy Products Through Our App!' on the Apple Dev Center
[11:23:39]: Your app name includes non-ASCII characters, which are not supported by the Apple Developer Portal.
[11:23:39]: To fix this a unique (internal) name 'Timeless Skin Care  Buy Products Through Our App dc5aef8ecd22e6fee078725770739611' has been created for you. Your app's real name 'Timeless Skin Care - Buy Products Through Our App!'
[11:23:39]: will still show up correctly on iTunes Connect and the App Store.
[11:23:39]: Created app CPH2J8Q2E5
[11:23:39]: Finished creating new app 'Timeless Skin Care - Buy Products Through Our App!' on the Dev Center
[11:23:42]: Creating new app 'Timeless Skin Care - Buy Products Through Our App!' on iTunes Connect
[11:23:44]: Ensuring version number
[11:23:45]: Successfully created new app 'Timeless Skin Care - Buy Products Through Our App!' on iTunes Connect with ID 1274205561
[11:23:45]: --------------------------------------------------------------------
[11:23:45]: Step: fastlane produce merchant -o merchant.com.tapcart.XXXXXXXX -r "iOS Apple Pay Merchant"
[11:23:45]: --------------------------------------------------------------------
[11:23:45]: $ fastlane produce merchant -o merchant.com.tapcart.XXXXXXXX -r "iOS Apple Pay Merchant"
[11:23:46]: ▸ [11:23:46]: For more information, check out https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[11:23:48]: ▸ [11:23:48]: Starting login with user 'xxx@tapcart.co'
[11:23:49]: ▸ [11:23:49]: Successfully logged in
[11:23:49]: ▸ [11:23:49]: Creating new merchant 'iOS Apple Pay Merchant' with identifier 'merchant.com.tapcart.XXXXXXXX' on the Apple Dev Center
[11:23:50]: ▸ [11:23:50]: Created merchant ZQZ6BGA25L
[11:23:50]: ▸ [11:23:50]: Finished creating new merchant 'iOS Apple Pay Merchant' on the Dev Center
[11:23:50]: --------------------------------------------------------------------
[11:23:50]: Step: fastlane produce associate_merchant -a com.tapcart.XXXXXXXX merchant.com.tapcart.XXXXXXXX
[11:23:50]: --------------------------------------------------------------------
[11:23:50]: $ fastlane produce associate_merchant -a com.tapcart.XXXXXXXX merchant.com.tapcart.XXXXXXXX
[11:23:51]: ▸ [11:23:51]: For more information, check out https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[11:23:53]: ▸ [11:23:53]: Starting login with user 'xxx@tapcart.co'
[11:23:55]: ▸ [11:23:55]: Successfully logged in
[11:23:55]: ▸ [11:23:55]: Validating merchants before association
[11:23:56]: ▸ [11:23:56]: Finalising association with 1 merchant
[11:23:57]: ▸ [11:23:57]: Done!
[11:23:57]: ---------------------
[11:23:57]: --- Step: produce ---
[11:23:57]: ---------------------
+----------------+-------------------------------------+
|              Summary for produce 2.53.1              |
+----------------+-------------------------------------+
| username       | xxx@tapcart.co                     |
| app_identifier | com.tapcart.XXXXXXXX.notification |
| app_name       | iOS Notification Extension          |
| skip_itc       | true                                |
| sku            | 1503426237                          |
| platform       | ios                                 |
| language       | English                             |
| skip_devcenter | false                               |
| team_id        | XXXXXXXX                          |
+----------------+-------------------------------------+
[11:23:58]: Creating new app 'iOS Notification Extension' on the Apple Dev Center
[11:23:59]: Created app G856C4KQZX
[11:23:59]: Finished creating new app 'iOS Notification Extension' on the Dev Center
[11:23:59]: -----------------------------
[11:23:59]: --- Step: create_keychain ---
[11:23:59]: -----------------------------
[11:23:59]: $ security list-keychains -d user
[11:23:59]: ▸ "/Users/vagrant/Library/Keychains/Tapcart-Release-db"
[11:23:59]: -------------------
[11:23:59]: --- Step: match ---
[11:23:59]: -------------------
+-----------------------+-------------------------------------------------------------------+
|                                 Summary for match 2.53.1                                  |
+-----------------------+-------------------------------------------------------------------+
| git_url               | git@github.com:tapcartinc/certs-profiles.git                      |
| git_branch            | XXXXXXXX                                                        |
| type                  | appstore                                                          |
| app_identifier        | ["com.tapcart.XXXXXXXX", "com.tapcart.XXXXXXXX.notification"] |
| git_full_name         | tapcart-bot                                                       |
| readonly              | false                                                             |
| git_user_email        | xxx@tapcart.co                                                   |
| username              | xxx@tapcart.co                                                   |
| shallow_clone         | true                                                              |
| keychain_name         | Tapcart-Release                                                   |
| keychain_password     | ********                                                          |
| force                 | true                                                              |
| team_id               | XXXXXXXX                                                        |
| verbose               | false                                                             |
| skip_confirmation     | false                                                             |
| clone_branch_directly | false                                                             |
| force_for_new_devices | false                                                             |
| skip_docs             | false                                                             |
| platform              | ios                                                               |
+-----------------------+-------------------------------------------------------------------+
[11:23:59]: Cloning remote git repo...
[11:23:59]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[11:24:00]: Add git user config to local git repo...
[11:24:00]: Checking out branch XXXXXXXX...
[11:24:00]: 🔓  Successfully decrypted certificates repo
[11:24:00]: Verifying that the certificate and profile are still valid on the Dev Portal...
[11:24:01]: Couldn't find a valid code signing identity in the git repo for distribution... creating one for you now
+---------------+-----------------------------------------------------+
|                       Summary for cert 2.53.1                       |
+---------------+-----------------------------------------------------+
| development   | false                                               |
| force         | true                                                |
| username      | xxx@tapcart.co                                     |
| team_id       | XXXXXXXX                                          |
| keychain_path | /Users/vagrant/Library/Keychains/Tapcart-Release-db |
| platform      | ios                                                 |
+---------------+-----------------------------------------------------+
[11:24:01]: Starting login with user 'xxx@tapcart.co'
[11:24:02]: Successfully logged in
[11:24:03]: Successfully generated 268T8K3T84 which was imported to the local machine.
security: SecKeychainSearchCopyNext: The specified item could not be found in the keychain.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  1062  100  1062    0     0   3123      0 --:--:-- --:--:-- --:--:--  3132
[11:24:04]: Verifying the certificate is properly installed locally...
[11:24:04]: Successfully installed certificate 268T8K3T84
+-------------------------------------+---------------------------------------+
|                           Summary for sigh 2.53.1                           |
+-------------------------------------+---------------------------------------+
| app_identifier                      | com.tapcart.XXXXXXXX                |
| username                            | xxx@tapcart.co                       |
| force                               | true                                  |
| cert_id                             | 268T8K3T84                            |
| provisioning_name                   | match AppStore com.tapcart.XXXXXXXX |
| ignore_profiles_with_different_name | true                                  |
| team_id                             | XXXXXXXX                            |
| platform                            | ios                                   |
| adhoc                               | false                                 |
| development                         | false                                 |
| skip_install                        | false                                 |
| skip_fetch_profiles                 | false                                 |
| skip_certificate_verification       | false                                 |
+-------------------------------------+---------------------------------------+
[11:24:04]: Starting login with user 'xxx@tapcart.co'
[11:24:05]: Successfully logged in
[11:24:05]: Fetching profiles...
[11:24:06]: Verifying certificates...
[11:24:06]: No existing profiles found, that match the certificates you have installed locally! Creating a new provisioning profile for you
[11:24:07]: Creating new provisioning profile for 'com.tapcart.XXXXXXXX' with name 'match AppStore com.tapcart.XXXXXXXX' for 'ios' platform
[11:24:08]: Downloading provisioning profile...
[11:24:08]: Successfully downloaded provisioning profile...
[11:24:08]: Installing provisioning profile...
/var/folders/90/5stft2v13fb_m_gv3c8x9nwc0000gn/T/d20170822-1699-12ifphz/profiles/appstore/AppStore_com.tapcart.XXXXXXXX.mobileprovision
[11:24:08]: Installing provisioning profile...
+-------------------------------------+----------------------------------------------------+
|                                 Summary for sigh 2.53.1                                  |
+-------------------------------------+----------------------------------------------------+
| app_identifier                      | com.tapcart.XXXXXXXX.notification                |
| username                            | xxx@tapcart.co                                    |
| force                               | true                                               |
| cert_id                             | 268T8K3T84                                         |
| provisioning_name                   | match AppStore com.tapcart.XXXXXXXX.notification |
| ignore_profiles_with_different_name | true                                               |
| team_id                             | XXXXXXXX                                         |
| platform                            | ios                                                |
| adhoc                               | false                                              |
| development                         | false                                              |
| skip_install                        | false                                              |
| skip_fetch_profiles                 | false                                              |
| skip_certificate_verification       | false                                              |
+-------------------------------------+----------------------------------------------------+
[11:24:09]: Starting login with user 'xxx@tapcart.co'
[11:24:10]: Successfully logged in
[11:24:10]: Fetching profiles...
[11:24:10]: Verifying certificates...
[11:24:10]: No existing profiles found, that match the certificates you have installed locally! Creating a new provisioning profile for you
[11:24:12]: Creating new provisioning profile for 'com.tapcart.XXXXXXXX.notification' with name 'match AppStore com.tapcart.XXXXXXXX.notification' for 'ios' platform
[11:24:12]: Downloading provisioning profile...
[11:24:13]: Successfully downloaded provisioning profile...
[11:24:13]: Installing provisioning profile...
/var/folders/90/5stft2v13fb_m_gv3c8x9nwc0000gn/T/d20170822-1699-12ifphz/profiles/appstore/AppStore_com.tapcart.XXXXXXXX.notification.mobileprovision
[11:24:13]: Installing provisioning profile...
[11:24:13]: 🔒  Successfully encrypted certificates repo
[11:24:13]: Pushing changes to remote git repo...
+---------------------+---------------------------------------------------+----------------------------------------------------------------------------------------------------------------+
|                                                                              Installed Provisioning Profile                                                                              |
+---------------------+---------------------------------------------------+----------------------------------------------------------------------------------------------------------------+
| Parameter           | Environment Variable                              | Value                                                                                                          |
+---------------------+---------------------------------------------------+----------------------------------------------------------------------------------------------------------------+
| App Identifier      |                                                   | com.tapcart.XXXXXXXX                                                                                         |
| Type                |                                                   | appstore                                                                                                       |
| Platform            |                                                   | ios                                                                                                            |
| Profile UUID        | sigh_com.tapcart.XXXXXXXX_appstore              | c7bdf25b-6bf5-42ca-bb8a-f5958951c3b8                                                                           |
| Profile Name        | sigh_com.tapcart.XXXXXXXX_appstore_profile-name | match AppStore com.tapcart.XXXXXXXX                                                                          |
| Profile Path        | sigh_com.tapcart.XXXXXXXX_appstore_profile-path | /Users/vagrant/Library/MobileDevice/Provisioning Profiles/c7bdf25b-6bf5-42ca-bb8a-f5958951c3b8.mobileprovision |
| Development Team ID | sigh_com.tapcart.XXXXXXXX_appstore_team-id      | XXXXXXXX                                                                                                     |
+---------------------+---------------------------------------------------+----------------------------------------------------------------------------------------------------------------+
+---------------------+----------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------+
|                                                                                    Installed Provisioning Profile                                                                                     |
+---------------------+----------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------+
| Parameter           | Environment Variable                                           | Value                                                                                                          |
+---------------------+----------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------+
| App Identifier      |                                                                | com.tapcart.XXXXXXXX.notification                                                                            |
| Type                |                                                                | appstore                                                                                                       |
| Platform            |                                                                | ios                                                                                                            |
| Profile UUID        | sigh_com.tapcart.XXXXXXXX.notification_appstore              | 243ce646-03cc-41f2-96d6-2fd99541b95e                                                                           |
| Profile Name        | sigh_com.tapcart.XXXXXXXX.notification_appstore_profile-name | match AppStore com.tapcart.XXXXXXXX.notification                                                             |
| Profile Path        | sigh_com.tapcart.XXXXXXXX.notification_appstore_profile-path | /Users/vagrant/Library/MobileDevice/Provisioning Profiles/243ce646-03cc-41f2-96d6-2fd99541b95e.mobileprovision |
| Development Team ID | sigh_com.tapcart.XXXXXXXX.notification_appstore_team-id      | XXXXXXXX                                                                                                     |
+---------------------+----------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------+
[11:24:15]: All required keys, certificates and provisioning profiles are installed 🙌
[11:24:15]: Setting Provisioning Profile type to 'app-store'
[11:24:15]: -----------------
[11:24:15]: --- Step: gym ---
[11:24:15]: -----------------
[11:24:16]: Detected provisioning profile mapping: {"com.tapcart.XXXXXXXX"=>"match AppStore com.tapcart.XXXXXXXX", "com.tapcart.XXXXXXXX.notification"=>"match AppStore com.tapcart.XXXXXXXX.notification"}
[11:24:16]: $ xcodebuild -list -workspace Tapcart.xcworkspace
[11:24:18]: $ xcodebuild -showBuildSettings -workspace Tapcart.xcworkspace -scheme Tapcart
2017-08-22 11:24:20.532 xcodebuild[1805:9794] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-12122/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/ProjectModel/Capabilities/Xcode3TargetCapabilitiesContext.m:320
Details:  Failed to expand identifiers "{(
    "merchant.com.tapcart.XXXXXXXX"
)}": Error Domain=IDECapabilitiesErrorDomain Code=1 "The operation requires a selected developer portal team." UserInfo={NSLocalizedDescription=The operation requires a selected developer portal team., NSLocalizedRecoverySuggestion=}
Object:   <Xcode3TargetCapabilitiesContext: 0x7fd6a959a770>
Method:   -_appIDFeatureStateIncludingEntitlements:buildConfiguration:
Thread:   <NSThread: 0x7fd6a74142b0>{number = 1, name = main}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
2017-08-22 11:24:20.547 xcodebuild[1805:9794] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-12122/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/ProjectModel/Capabilities/Xcode3TargetCapabilitiesContext.m:320
Details:  Failed to expand identifiers "{(
    "merchant.com.tapcart.XXXXXXXX"
)}": Error Domain=IDECapabilitiesErrorDomain Code=1 "The operation requires a selected developer portal team." UserInfo={NSLocalizedDescription=The operation requires a selected developer portal team., NSLocalizedRecoverySuggestion=}
Object:   <Xcode3TargetCapabilitiesContext: 0x7fd6a959a770>
Method:   -_appIDFeatureStateIncludingEntitlements:buildConfiguration:
Thread:   <NSThread: 0x7fd6a74142b0>{number = 1, name = main}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
+-------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
|                                                                                          Summary for gym 2.53.1                                                                                           |
+-------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
| workspace                                                               | Tapcart.xcworkspace                                                                                                             |
| scheme                                                                  | Tapcart                                                                                                                         |
| silent                                                                  | false                                                                                                                           |
| clean                                                                   | true                                                                                                                            |
| output_name                                                             | Tapcart                                                                                                                         |
| export_method                                                           | app-store                                                                                                                       |
| xcargs                                                                  | TEAM=XXXXXXXX MAIN_PROF=match\ AppStore\ com.tapcart.XXXXXXXX EXT_PROF=match\ AppStore\ com.tapcart.XXXXXXXX.notification |
| export_options.provisioningProfiles.com.tapcart.XXXXXXXX              | match AppStore com.tapcart.XXXXXXXX                                                                                           |
| export_options.provisioningProfiles.com.tapcart.XXXXXXXX.notification | match AppStore com.tapcart.XXXXXXXX.notification                                                                              |
| destination                                                             | generic/platform=iOS                                                                                                            |
| build_path                                                              | /Users/vagrant/Library/Developer/Xcode/Archives/2017-08-22                                                                      |
| output_directory                                                        | .                                                                                                                               |
| skip_package_ipa                                                        | false                                                                                                                           |
| buildlog_path                                                           | /var/folders/90/5stft2v13fb_m_gv3c8x9nwc0000gn/T/fastlane_logs235528095/gym                                                     |
| xcode_path                                                              | /Applications/Xcode.app                                                                                                         |
+-------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
[11:24:20]: $ set -o pipefail && xcodebuild -workspace Tapcart.xcworkspace -scheme Tapcart -destination 'generic/platform=iOS' -archivePath /Users/vagrant/Library/Developer/Xcode/Archives/2017-08-22/Tapcart\ 2017-08-22\ 11.24.20.xcarchive TEAM=XXXXXXXX MAIN_PROF=match\ AppStore\ com.tapcart.XXXXXXXX EXT_PROF=match\ AppStore\ com.tapcart.XXXXXXXX.notification clean archive | tee /var/folders/90/5stft2v13fb_m_gv3c8x9nwc0000gn/T/fastlane_logs235528095/gym/Tapcart-Tapcart.log | xcpretty
[11:24:24]: ▸ 2017-08-22 11:24:24.211 xcodebuild[1813:9839] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-12122/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/ProjectModel/Capabilities/Xcode3TargetCapabilitiesContext.m:320
[11:24:24]: ▸ Details:  Failed to expand identifiers "{(
[11:24:24]: ▸ "merchant.com.tapcart.XXXXXXXX"
[11:24:24]: ▸ )}": Error Domain=IDECapabilitiesErrorDomain Code=1 "The operation requires a selected developer portal team." UserInfo={NSLocalizedDescription=The operation requires a selected developer portal team., NSLocalizedRecoverySuggestion=}
[11:24:24]: ▸ Object:   <Xcode3TargetCapabilitiesContext: 0x7fd6dafe0f40>
[11:24:24]: ▸ Method:   -_appIDFeatureStateIncludingEntitlements:buildConfiguration:
[11:24:24]: ▸ Thread:   <NSThread: 0x7fd6d8515530>{number = 1, name = main}
[11:24:24]: ▸ Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
[11:24:24]: ▸ 2017-08-22 11:24:24.219 xcodebuild[1813:9839] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-12122/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/ProjectModel/Capabilities/Xcode3TargetCapabilitiesContext.m:320
[11:24:24]: ▸ Details:  Failed to expand identifiers "{(
[11:24:24]: ▸ "merchant.com.tapcart.XXXXXXXX"
[11:24:24]: ▸ )}": Error Domain=IDECapabilitiesErrorDomain Code=1 "The operation requires a selected developer portal team." UserInfo={NSLocalizedDescription=The operation requires a selected developer portal team., NSLocalizedRecoverySuggestion=}
[11:24:24]: ▸ Object:   <Xcode3TargetCapabilitiesContext: 0x7fd6dafe0f40>
[11:24:24]: ▸ Method:   -_appIDFeatureStateIncludingEntitlements:buildConfiguration:
[11:24:24]: ▸ Thread:   <NSThread: 0x7fd6d8515530>{number = 1, name = main}
[11:24:24]: ▸ Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
[11:24:24]: ▸ Cleaning Tapcart/notification [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/Bolts [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/Caishen [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/ChameleonFramework [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/CountryPickerSwift-CountryPicker [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/CountryPickerSwift [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/IOStickyHeader [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/Kingfisher [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/Mobile-Buy-SDK-Buy [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/NVActivityIndicatorView [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/Mobile-Buy-SDK [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:24]: ▸ Cleaning Pods/Parse [Debug]
[11:24:24]: ▸ Check Dependencies
[11:24:25]: ▸ Cleaning Pods/PopupDialog [Debug]
[11:24:25]: ▸ Check Dependencies
[11:24:25]: ▸ Cleaning Pods/SwiftyJSON [Debug]
[11:24:25]: ▸ Check Dependencies
[11:24:25]: ▸ Cleaning Pods/SwiftyOnboard [Debug]
[11:24:25]: ▸ Check Dependencies
[11:24:25]: ▸ Cleaning Pods/UICircularProgressRing [Debug]
[11:24:25]: ▸ Check Dependencies
[11:24:25]: ▸ Cleaning Pods/Pods-Tapcart [Debug]
[11:24:25]: ▸ Check Dependencies
[11:24:25]: ▸ Cleaning Tapcart/Tapcart [Debug]
[11:24:25]: ▸ Check Dependencies
[11:24:25]: ▸ 2017-08-22 11:24:25.716 xcodebuild[1813:9839] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-12122/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/ProjectModel/Capabilities/Xcode3TargetCapabilitiesContext.m:320
[11:24:25]: ▸ Details:  Failed to expand identifiers "{(
[11:24:25]: ▸ "merchant.com.tapcart.XXXXXXXX"
[11:24:25]: ▸ )}": Error Domain=IDECapabilitiesErrorDomain Code=1 "The operation requires a selected developer portal team." UserInfo={NSLocalizedDescription=The operation requires a selected developer portal team., NSLocalizedRecoverySuggestion=}
[11:24:25]: ▸ Object:   <Xcode3TargetCapabilitiesContext: 0x7fd6dafe0f40>
[11:24:25]: ▸ Method:   -_appIDFeatureStateIncludingEntitlements:buildConfiguration:
[11:24:25]: ▸ Thread:   <NSThread: 0x7fd6d8515530>{number = 1, name = main}
[11:24:25]: ▸ Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
[11:24:25]: ▸ 2017-08-22 11:24:25.719 xcodebuild[1813:9839] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-12122/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/ProjectModel/Capabilities/Xcode3TargetCapabilitiesContext.m:320
[11:24:25]: ▸ Details:  Failed to expand identifiers "{(
[11:24:25]: ▸ "merchant.com.tapcart.XXXXXXXX"
[11:24:25]: ▸ )}": Error Domain=IDECapabilitiesErrorDomain Code=1 "The operation requires a selected developer portal team." UserInfo={NSLocalizedDescription=The operation requires a selected developer portal team., NSLocalizedRecoverySuggestion=}
[11:24:25]: ▸ Object:   <Xcode3TargetCapabilitiesContext: 0x7fd6dafe0f40>
[11:24:25]: ▸ Method:   -_appIDFeatureStateIncludingEntitlements:buildConfiguration:
[11:24:25]: ▸ Thread:   <NSThread: 0x7fd6d8515530>{number = 1, name = main}
[11:24:25]: ▸ Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
[11:24:25]: ▸ Clean Succeeded
[11:24:26]: ▸ 2017-08-22 11:24:26.001 xcodebuild[1813:9839] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-12122/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/ProjectModel/Capabilities/Xcode3TargetCapabilitiesContext.m:320
[11:24:26]: ▸ Details:  Failed to expand identifiers "{(
[11:24:26]: ▸ "merchant.com.tapcart.XXXXXXXX"
[11:24:26]: ▸ )}": Error Domain=IDECapabilitiesErrorDomain Code=1 "The operation requires a selected developer portal team." UserInfo={NSLocalizedDescription=The operation requires a selected developer portal team., NSLocalizedRecoverySuggestion=}
[11:24:26]: ▸ Object:   <Xcode3TargetCapabilitiesContext: 0x7fd6dafe0f40>
[11:24:26]: ▸ Method:   -_appIDFeatureStateIncludingEntitlements:buildConfiguration:
[11:24:26]: ▸ Thread:   <NSThread: 0x7fd6d8515530>{number = 1, name = main}
[11:24:26]: ▸ Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
[11:24:26]: ▸ 2017-08-22 11:24:26.005 xcodebuild[1813:9839] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEXcode3ProjectSupport/IDEXcode3ProjectSupport-12122/Xcode3Core/LegacyProjects/Frameworks/DevToolsCore/DevToolsCore/ProjectModel/Capabilities/Xcode3TargetCapabilitiesContext.m:320
[11:24:26]: ▸ Details:  Failed to expand identifiers "{(
[11:24:26]: ▸ "merchant.com.tapcart.XXXXXXXX"
[11:24:26]: ▸ )}": Error Domain=IDECapabilitiesErrorDomain Code=1 "The operation requires a selected developer portal team." UserInfo={NSLocalizedDescription=The operation requires a selected developer portal team., NSLocalizedRecoverySuggestion=}
[11:24:26]: ▸ Object:   <Xcode3TargetCapabilitiesContext: 0x7fd6dafe0f40>
[11:24:26]: ▸ Method:   -_appIDFeatureStateIncludingEntitlements:buildConfiguration:
[11:24:26]: ▸ Thread:   <NSThread: 0x7fd6d8515530>{number = 1, name = main}
[11:24:26]: ▸ Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
[11:24:26]: ▸ Building Tapcart/notification [Release]
[11:24:26]: ▸ Check Dependencies
[11:24:26]: ▸ Processing Info.plist
[11:24:26]: ▸ Compiling NotificationService.swift
[11:24:32]: ▸ Compiling NotificationService.swift
[11:24:35]: ▸ Linking notification
[11:24:37]: ▸ Linking notification
[11:24:38]: ▸ Generating 'notification.appex.dSYM'
[11:24:39]: ▸ Touching notification.appex
[11:24:39]: ▸ Signing /Users/vagrant/Library/Developer/Xcode/DerivedData/Tapcart-cevtwioepdxzkngbhhtyqcxrmepg/Build/Intermediates/ArchiveIntermediates/Tapcart/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/notification.appex

```

**Comment by EricNetsch** [comment]
I have temporarily gotten around this by running Match first in a previous pipeline before re-running it a second time while deploying the app. Far from an ideal solution. Really curious if anyone else had had this issue with gym getting stuck signing

Any ideas? @cbowns @taquitos

**Comment by mpirri** [comment]
@EricNetsch Thanks for writing in! Very interesting that `gym` just hangs up in this case. Have you tried running fastlane with the "verbose" flag set (e.g., `fastlane <lane> --verbose`) to see if that gives more output that helps explain why `gym` is just hanging?

**Comment by EricNetsch** [comment]
@mpirri  Still haven't been able to solve this, driving me nuts these last few weeks. Probably not a big deal for most people using only a few dev accounts. I'll be switching everything to xcode 9 which I'm hoping will resolve this issue. I will close this out once I get past this. Stay tuned

**Comment by SuperTango** [comment]
I'm seeing this same behavior.  

[22:12:19]: ▸ Running script '[CP] Copy Pods Resources'
[22:12:23]: ▸ Touching JamImessageExtension.appex
[22:12:25]: ▸ Signing /Users/distiller/Library/Developer/Xcode/DerivedData/Playlist-aqtngtvmjobmrfgiexeqvibhsmyr/Build/Intermediates.noindex/ArchiveIntermediates/CI_Build/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/JamImessageExtension.appex
(hang here)

**Comment by SuperTango** [comment]
I ran fastlane  with --verbose, but no additional data was provided.

**Comment by mpirri** [comment]
So, a freeze on `gym` while running on a CI situation leads me to wonder if perhaps Xcode is trying to prompt you with the "Enter your password to unlock the keychain" message. Have you tried putting a call to the `unlock_keychain` action before running gym. It seems weird to require this if you only created the keychain for this run, but definitely worth trying if you haven't already? 🚀

**Comment by eyalabadi98** [comment]
I'm having the same issue with Signing stuck on Signing for over 20 minutes... Following

**Comment by ohayon** [comment]
Hey @eyalabadi98 - did you see if maybe you were victim to the issue that @mpirri pointed out above? That seems like it is likely the case 😄

**Comment by eyalabadi98** [comment]
I think the issue was that my Xctest was failing. I had to go on xcode and edit my test to run by changing the text is has to see in order to be successful.

**Comment by ohayon** [comment]
@eyalabadi98 could you share the code that youre using to call fastlane?

**Comment by eyalabadi98** [comment]
When I use fastlane ios beta. It gets stuck at different places almost every time and always ends up freezing and slowing down my computer

**Comment by ohayon** [comment]
Apologies for the issue @eyalabadi98 - let us know if you can share the full output so we can see whats going on. Thanks!! 🐰

**Comment by eyalabadi98** [comment]
@ohayon attached is the output: https://pastebin.com/3euAf3mq

**Comment by ohayon** [comment]
Thanks @eyalabadi98! Could you try running just `xcodebuild -scheme MG16 -project ./MG16.xcodeproj -derivedDataPath '/var/folders/sp/587t8y_s0dsfc08kff7mpvr80000gn/T/snapshot_derived20171031-8330-qqu4te' -destination 'platform=iOS Simulator,name=iPhone 7,OS=11.0.1' -destination 'platform=iOS Simulator,name=iPhone 7 Plus,OS=11.0.1' FASTLANE_SNAPSHOT=YES build test` and see if that stalls the same way?

**Comment by EricNetsch** [comment]
Seems like this issue has resolved itself for newer versions of Fastlane. Feel free to close this if everyone agrees.

**Comment by ohayon** [comment]
Woot! Thanks for the update @EricNetsch. Ill close this for now, let us know if anyone is still having trouble! 🚀

**Comment by JanBrinker** [comment]
I'm currently experiencing this issue with fastlane 2.68.2, but only on our CircleCI and only after migrating to CircleCI 2.0, while leaving the lanes as they were with CircleCI 1.0.

The lanes run through on my machine and they used to run through on CircleCI 1.0. But I'm getting the same behavior as the others: hanging on "Signing ... *.appex"

**Comment by ohayon** [comment]
Hmmm. Sorry about the issue @JanBrinker - have you filed a bug with CircleCI at all? Its definitely something we should figure out, just curious if they have given any insight. Thanks! 🚀

**Comment by JanBrinker** [comment]
Hey @ohayon :) I have tried opening a thread on their support board, but that didn't result in anything. So now I filed a real bug ticket with them. I also spent a bit of time trying to figure out what I could have possibly done wrong.

My conclusion is: it has to have something to do with their new build environment for 2.0. The exact same lanes run through smoothly on 1.0. I suspect that for some reason the keychain is not unlocked properly, however why then can `match` sync the certs and profiles without hiccup? Mysterious. Let's wait for the response of CircleCI!

**Comment by nfeld9807** [comment]
@JanBrinker I'm discovering the same issue with CircleCI hanging on 2.0 builds. If you hear back, can you post an update? Below is a screenshot if anyone else is running into this issue with them as well.
<img width="1421" alt="screen shot 2017-12-11 at 4 25 37 pm" src="https://user-images.githubusercontent.com/1986818/33854505-f5bc3160-de8f-11e7-8f84-43f5141add6e.png">

**Comment by JanBrinker** [comment]
@nfeld9807 I will update this post when I hear back. However I think it would be good to signal CircleCI that this is not an isolated issue. So it would be great if you could do the following two things:

- Post your issue on https://discuss.circleci.com/t/issues-with-code-signing-on-circleci-2-0/18642
- Send a bug report to CircleCI

This might help us getting the folks at CircleCI fix this a bit quicker.

**Comment by JanBrinker** [comment]
@nfeld9807 the CircleCI team handed us the solution. You simply need to add one command in your lane or in before_all: `setup_circle_ci`

Yes, that is a ready made fastlane action for setting up the keychain etc specially for CircleCI. This command is *really* easy to overlook in their docs. But adding it to our Fastfile did the trick!

**Comment by nfeld9807** [comment]
@JanBrinker awesome, I will give that a shot, thanks

**Comment by rafalszastok** [comment]
I'm facing this issue again. I had this issue on December, but after update it was resolved by itself. Now I'm facing the same issue. Signing hangs for over 20 minutes.
